src/client.rs

```diff
@@ -152,8 +177,22 @@ impl ClientFlow {
                 return Ok(event);
             }
 
-            if let Some(event) = self.progress_receive().await? {
-                return Ok(event);
+            if self.is_waiting_for_idle_done_set() {
+                let timeout =
+                    tokio::time::timeout(self.get_idle_timeout(), self.progress_receive());
+                match timeout.await {
```

Comment:
Please note that the `Future` here can be cancelled. E.g. our proxy implementation cancels the `progress` `Future` all the time because it uses tokio's `select` macro. Then the timer would be started again with the whole timeout interval.

Reply:
Do I have to do sth specific here to handle future cancellation?

Reply:
I think there is no general solution for that. In my case It was sufficient to avoid local variables when reaching an `.await` point. I saved the local state somewhere inside of `ClientFlow` or `ServerFlow`. But your case is more complicated because time is involved.

However, with SansIO `ClientFlow` or `ServerFlow` won't use any `Future`s anymore. So let's wait and think about a solution later.